From ecb54c0ef7b36aab452ed456f621a972d49c6d3a Mon Sep 17 00:00:00 2001
From: Jan Malek <jmalek@internships.antmicro.com>
Date: Tue, 30 Aug 2022 13:11:20 +0200
Subject: [PATCH] Redirect console to UART

Boards: 96b_avenger96, hexiwear_kw40z, stm32h747i_disco and stm32mp157c_dk2
couldn't be build on CI, because we require a console output on UART for all boards
and by default their console output was directed elsewhere.
This changes the dts files of the mentioned boards to redirect their console output to
UART and allow them to be built and tested in Renode by the CI
---
 boards/arm/96b_avenger96/96b_avenger96.dts                | 2 ++
 boards/arm/96b_avenger96/96b_avenger96_defconfig          | 6 +++---
 boards/arm/hexiwear_kw40z/hexiwear_kw40z.dts              | 2 ++
 boards/arm/stm32h747i_disco/stm32h747i_disco_m4.dts       | 4 +++-
 boards/arm/stm32h747i_disco/stm32h747i_disco_m4_defconfig | 4 ++--
 boards/arm/stm32mp157c_dk2/stm32mp157c_dk2.dts            | 2 ++
 boards/arm/stm32mp157c_dk2/stm32mp157c_dk2_defconfig      | 6 +++---
 7 files changed, 17 insertions(+), 9 deletions(-)

diff --git a/boards/arm/96b_avenger96/96b_avenger96.dts b/boards/arm/96b_avenger96/96b_avenger96.dts
index 3accaae12a..533f174d25 100644
--- a/boards/arm/96b_avenger96/96b_avenger96.dts
+++ b/boards/arm/96b_avenger96/96b_avenger96.dts
@@ -22,6 +22,8 @@
 		 * zephyr,console = &uart7;
 		 * zephyr,shell-uart = &uart7;
 		 */
+		zephyr,console = &uart7;
+		zephyr,shell-uart = &uart7;
 		zephyr,flash = &retram;
 		zephyr,sram = &mcusram;
 	};
diff --git a/boards/arm/96b_avenger96/96b_avenger96_defconfig b/boards/arm/96b_avenger96/96b_avenger96_defconfig
index 05be01fbb3..d48a8dbae6 100644
--- a/boards/arm/96b_avenger96/96b_avenger96_defconfig
+++ b/boards/arm/96b_avenger96/96b_avenger96_defconfig
@@ -19,11 +19,11 @@ CONFIG_CLOCK_CONTROL=y
 
 # console (remote proc console by default)
 CONFIG_CONSOLE=y
-CONFIG_RAM_CONSOLE=y
-CONFIG_RAM_CONSOLE_BUFFER_SIZE=1024
+#CONFIG_RAM_CONSOLE=y
+#CONFIG_RAM_CONSOLE_BUFFER_SIZE=1024
 
 # uart console (overrides remote proc console)
-CONFIG_UART_CONSOLE=n
+CONFIG_UART_CONSOLE=y
 
 # enable pin controller
 CONFIG_PINCTRL=y
diff --git a/boards/arm/hexiwear_kw40z/hexiwear_kw40z.dts b/boards/arm/hexiwear_kw40z/hexiwear_kw40z.dts
index 2b64e0bafd..9ef32696bd 100644
--- a/boards/arm/hexiwear_kw40z/hexiwear_kw40z.dts
+++ b/boards/arm/hexiwear_kw40z/hexiwear_kw40z.dts
@@ -10,6 +10,8 @@
 	compatible = "nxp,kw40z", "nxp,mkw40z4";
 
 	chosen {
+        zephyr,console = &lpuart0;
+		zephyr,shell-uart = &lpuart0;
 		zephyr,sram = &sram0;
 		zephyr,flash = &flash0;
 	};
diff --git a/boards/arm/stm32h747i_disco/stm32h747i_disco_m4.dts b/boards/arm/stm32h747i_disco/stm32h747i_disco_m4.dts
index 74681a8829..00078ec23b 100644
--- a/boards/arm/stm32h747i_disco/stm32h747i_disco_m4.dts
+++ b/boards/arm/stm32h747i_disco/stm32h747i_disco_m4.dts
@@ -17,6 +17,8 @@
 	chosen {
 		/* zephyr,console = &usart1; */
 		/* zephyr,shell-uart = &usart1; */
+		zephyr,console = &usart1;
+		zephyr,shell-uart = &usart1;
 		zephyr,sram = &sram1;
 		zephyr,flash = &flash1;
 	};
@@ -48,7 +50,7 @@
 };
 
 &usart1 {
-	/* status = "okay"; */
+	status = "okay";
 };
 
 &uart8 {
diff --git a/boards/arm/stm32h747i_disco/stm32h747i_disco_m4_defconfig b/boards/arm/stm32h747i_disco/stm32h747i_disco_m4_defconfig
index 2f437301c8..045743dfb3 100644
--- a/boards/arm/stm32h747i_disco/stm32h747i_disco_m4_defconfig
+++ b/boards/arm/stm32h747i_disco/stm32h747i_disco_m4_defconfig
@@ -22,8 +22,8 @@ CONFIG_HW_STACK_PROTECTION=y
 CONFIG_SERIAL=y
 
 # By default CONSOLE is assigned to m7
-#CONFIG_CONSOLE=y
-#CONFIG_UART_CONSOLE=y
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
 
 # enable pin controller
 CONFIG_PINCTRL=y
diff --git a/boards/arm/stm32mp157c_dk2/stm32mp157c_dk2.dts b/boards/arm/stm32mp157c_dk2/stm32mp157c_dk2.dts
index 2704451697..339ed78d16 100644
--- a/boards/arm/stm32mp157c_dk2/stm32mp157c_dk2.dts
+++ b/boards/arm/stm32mp157c_dk2/stm32mp157c_dk2.dts
@@ -22,6 +22,8 @@
 		 * zephyr,console = &usart3;
 		 * zephyr,shell-uart = &usart3;
 		 */
+		zephyr,console = &usart3;
+		zephyr,shell-uart = &usart3;
 		zephyr,flash = &retram;
 		zephyr,sram = &mcusram;
 	};
diff --git a/boards/arm/stm32mp157c_dk2/stm32mp157c_dk2_defconfig b/boards/arm/stm32mp157c_dk2/stm32mp157c_dk2_defconfig
index 66503dc240..7962e99399 100644
--- a/boards/arm/stm32mp157c_dk2/stm32mp157c_dk2_defconfig
+++ b/boards/arm/stm32mp157c_dk2/stm32mp157c_dk2_defconfig
@@ -19,11 +19,11 @@ CONFIG_CLOCK_CONTROL=y
 
 # console (remote proc console by default)
 CONFIG_CONSOLE=y
-CONFIG_RAM_CONSOLE=y
-CONFIG_RAM_CONSOLE_BUFFER_SIZE=1024
+#CONFIG_RAM_CONSOLE=y
+#CONFIG_RAM_CONSOLE_BUFFER_SIZE=1024
 
 # uart console (overrides remote proc console)
-CONFIG_UART_CONSOLE=n
+CONFIG_UART_CONSOLE=y
 
 # enable pin controller
 CONFIG_PINCTRL=y
-- 
2.30.2

